pkg/mic/mic.go

```diff
@@ -223,9 +239,36 @@ func (c *Client) Sync(exit <-chan struct{}) {
 		if deleteList != nil && len(*deleteList) > 0 {
 			beginDeletion := time.Now()
 			workDone = true
+
+			vmssNodes := make(map[string][]*corev1.Node)
```

Comment:
Can we make it more generic ? or check if its only vmss then perform this ?

Reply:
So for each item in the list we need to check if the reference is against a vmss node and if any other node in the vmss is referencing it. This means we need to determine for each item if the node is a vmss.
Instead of doing this on demand I'm just gathering all the vmss nodes (which are being referenced) in one shot which prevents duplicate lookups.

In fact thinking about this more, this may not be enough... we may want to move adds before deletes so that we don't unnecessarily remove/re-add to a vmss node.